casper/src/main/scala/coop/rchain/casper/util/comm/ApproveBlockProtocol.scala

```diff
@@ -0,0 +1,131 @@
+package coop.rchain.casper.util.comm
+
+import com.google.protobuf.ByteString
+
+import cats.Monad
+import cats.effect.Sync
+import cats.effect.concurrent.Ref
+import cats.implicits._
+
+import coop.rchain.casper.{PrettyPrinter, Validate}
+import coop.rchain.casper.protocol._
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.hash.Blake2b256
+import coop.rchain.comm.CommError.ErrorHandler
+import coop.rchain.comm.PeerNode
+import coop.rchain.comm.discovery.NodeDiscovery
+import coop.rchain.comm.protocol.rchain.Packet
+import coop.rchain.comm.transport
+import coop.rchain.comm.transport.TransportLayer
+import coop.rchain.shared._
+import coop.rchain.catscontrib.{Capture, IOUtil}
+
+import monix.execution.atomic.AtomicAny
+
+import scala.util.Try
+
+class ApproveBlockProtocol[
+    F[_]: Capture: Sync: NodeDiscovery: TransportLayer: Log: Time: ErrorHandler] private (
+    val block: BlockMessage,
+    val requiredSigs: Int,
+    val start: Long,
+    val duration: Long,
+    private val sigsF: Ref[F, Set[Signature]]) {
+  private implicit val logSource: LogSource = LogSource(this.getClass)
+
+  val candidate                         = ApprovedBlockCandidate(Some(block), requiredSigs)
```

Comment:
If you add `[(scalapb.field).no_box = true]` to field in proto file you won't have them wrapped in `Option`.

Reply:
If you want I can add it as a part of the PR.

Reply:
Oh, neat. Yeah, that sounds useful.